apps/web/src/components/Layout/MaintenanceBanner.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import isAfter from "date-fns/isAfter";
+import isBefore from "date-fns/isBefore";
+import isEqual from "date-fns/isEqual";
+
+import {
+  currentDate,
+  parseDateTimeUtc,
+  relativeClosingDate,
+} from "@gc-digital-talent/date-helpers";
+import { getRuntimeVariable } from "@gc-digital-talent/env";
+import { Alert } from "@gc-digital-talent/ui";
+
+const MaintenanceBanner = () => {
+  const intl = useIntl();
+  // Outage time range, date, starttime
+  const maintenanceBannerPublicDate = getRuntimeVariable(
+    "MAINTENANCE_BANNER_PUBLIC_DATE",
+  );
+  const serverMaintenanceDate = getRuntimeVariable(
+    "MAINTENANCE_BANNER_START_DATE",
+  );
+  const serverMaintenanceDuration = getRuntimeVariable(
+    "MAINTENANCE_BANNER_DURATION",
+  );
+
+  console.log(serverMaintenanceDate);
+  const showMaintenanceBanner =
+    maintenanceBannerPublicDate && serverMaintenanceDate
+      ? (isEqual(
+          parseDateTimeUtc(maintenanceBannerPublicDate),
+          parseDateTimeUtc(currentDate()),
+        ) ||
+          isAfter(
+            parseDateTimeUtc(maintenanceBannerPublicDate),
+            parseDateTimeUtc(currentDate()),
+          )) &&
+        isBefore(
+          parseDateTimeUtc(maintenanceBannerPublicDate),
+          parseDateTimeUtc(serverMaintenanceDate),
+        )
+      : false;
+  return showMaintenanceBanner &&
+    serverMaintenanceDate &&
+    serverMaintenanceDuration ? (
+    <Alert.Root type="warning" live banner>
+      <Alert.Title>
+        {intl.formatMessage({
+          defaultMessage: "Scheduled server maintenance",
+          id: "H0RLNn",
+          description: "Heading for a server maintenance banner.",
+        })}
+      </Alert.Title>
+      <p>
+        {intl.formatMessage(
```

Comment:
Oops, watch out for the scenarios where the formatting function has special text.  I'd suggest just using one of the date-fns functions directly
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/a009cde5-aba1-4b1e-9c5d-b6def4428274)
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/062defbf-c0de-4e40-91ed-5d2b5aab1ae3)

Reply:
Nice catch again, should be fixed by [88f4c06](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7832/commits/88f4c06c3867f6804a4f87b8a770c5ce0b996bfd)

Should I leave in the possibility of decimal numbers? or just round up using `Math.ceil()`

Reply:
Leaving as a text is probably fine.  We could put in integers, decimals, or even things like "1-2".

Reply:
Oh, it only works for numbers.  I think that's fine. :+1: